Reduce use of raw pointers in containers in WebCore/

cdumez opened the pull request:
#### 80254cf0d73f6dc98199ff70bc0769f4e801add4
<pre>
Reduce use of raw pointers in containers in WebCore/
<a href="https://bugs.webkit.org/show_bug.cgi?id=302690">https://bugs.webkit.org/show_bug.cgi?id=302690</a>

Reviewed by Darin Adler and Geoffrey Garen.

* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::WebSocket::allActiveWebSockets):
(): Deleted.
* Source/WebCore/Modules/websockets/WebSocket.h:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
(WebCore::WTF_REQUIRES_LOCK):
(WebCore::ScriptExecutionContext::regenerateIdentifier):
(WebCore::ScriptExecutionContext::addToContextsMap):
(WebCore::ScriptExecutionContext::checkConsistency const):
(WebCore::ScriptExecutionContext::dispatchMessagePortEvents):
(WebCore::ScriptExecutionContext::createdMessagePort):
(WebCore::ScriptExecutionContext::destroyedMessagePort):
(WebCore::ScriptExecutionContext::registerServiceWorker):
* Source/WebCore/dom/ScriptExecutionContext.h:
* Source/WebCore/dom/ScriptExecutionContextInlines.h:
(WebCore::ScriptExecutionContext::serviceWorker): Deleted.
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::InspectorNetworkAgent::enable):
(WebCore::InspectorNetworkAgent::webSocketForRequestId):
(WebCore::webSocketAsScriptValue):
(WebCore::InspectorNetworkAgent::resolveWebSocket):
* Source/WebCore/inspector/agents/InspectorNetworkAgent.h:
* Source/WebCore/inspector/agents/page/PageNetworkAgent.cpp:
(WebCore::PageNetworkAgent::activeWebSockets):
(): Deleted.
* Source/WebCore/inspector/agents/page/PageNetworkAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerNetworkAgent.cpp:
(WebCore::WorkerNetworkAgent::activeWebSockets):
(): Deleted.
* Source/WebCore/inspector/agents/worker/WorkerNetworkAgent.h:
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::scriptExecutionContextIdentifierToLoaderMap):
(WebCore::DocumentLoader::fromScriptExecutionContextIdentifier):
(WebCore::DocumentLoader::loadMainResource):
(): Deleted.
* Source/WebCore/loader/archive/Archive.cpp:
(WebCore::Archive::clearAllSubframeArchives):
* Source/WebCore/loader/archive/Archive.h:
* Source/WebCore/page/PageGroup.cpp:
(WebCore::pageGroups):
(WebCore::PageGroup::pageGroup):
* Source/WebCore/workers/service/ServiceWorker.h:
(WebCore::ScriptExecutionContext::serviceWorker):

Canonical link: <a href="https://commits.webkit.org/303210@main">https://commits.webkit.org/303210@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/898c27449411b4e0ebe3f9a9066263d94eb02736

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131560 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3892 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42525 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/139068 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/83345 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e452f1be-ebec-4a50-9549-bd086a5d4234) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133430 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3982 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3732 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/100436 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/68026 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/afe685a5-72e6-4c02-abe9-e1151f3e2c42) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134506 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2847 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117757 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/81246 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/32f1c0f6-fbad-408f-bf34-2554ba2d3e07) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/2763 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/473 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82260 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/111389 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35881 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141714 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3636 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/36399 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108804 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3684 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/3237 "Found 1 new test failure: imported/w3c/web-platform-tests/webvtt/rendering/cues-with-video/processing-model/selectors/cue/white-space_normal_wrapped.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/109049 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27656 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2754 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/114086 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56830 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3697 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32490 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3524 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/67108 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3779 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3627 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/dom/ScriptExecutionContext.cpp:
```diff
@@ -253,14 +253,10 @@ void ScriptExecutionContext::dispatchMessagePortEvents()
-        if (m_messagePorts.contains(messagePort.get()) && messagePort->started())
-            messagePort->dispatchMessages();
-    }
+    m_messagePorts.forEach([](auto& messagePort) {
+        if (messagePort.started())
+            messagePort.dispatchMessages();
+    });
```
What makes it now safe to iterate this while ports might be added or removed?

geoffreygaren replied:
Perhaps idiosyncratically, WeakHashSet copies into a temporary vector when iterating.

darinadler replied:
Seems completely reasonable that it does that, but not good that nothing in the function name makes that explicit. I think we’d want to encode such behaviors into the names we give functions

cdumez replied:
To be clear, `for (auto& t : weakHashSet)` doesn't make a copy but `WeakHashSet::forEach()` does and I think this is why we have this function. To your point, the name doesn't make it clear. I'd be happy to rename this function in a separate PR if we find a good name. Maybe `WeakHashSet::safeIterate()`, `WeakHashSet::iterateOnCopy()`, `WeakHashSet::safeForEach()`

cdumez committed:
Reduce use of raw pointers in containers in WebCore/
https://bugs.webkit.org/show_bug.cgi?id=302690

Reviewed by Darin Adler and Geoffrey Garen.

* Source/WebCore/Modules/websockets/WebSocket.cpp:
(WebCore::WebSocket::allActiveWebSockets):
(): Deleted.
* Source/WebCore/Modules/websockets/WebSocket.h:
* Source/WebCore/dom/ScriptExecutionContext.cpp:
(WebCore::WTF_REQUIRES_LOCK):
(WebCore::ScriptExecutionContext::regenerateIdentifier):
(WebCore::ScriptExecutionContext::addToContextsMap):
(WebCore::ScriptExecutionContext::checkConsistency const):
(WebCore::ScriptExecutionContext::dispatchMessagePortEvents):
(WebCore::ScriptExecutionContext::createdMessagePort):
(WebCore::ScriptExecutionContext::destroyedMessagePort):
(WebCore::ScriptExecutionContext::registerServiceWorker):
* Source/WebCore/dom/ScriptExecutionContext.h:
* Source/WebCore/dom/ScriptExecutionContextInlines.h:
(WebCore::ScriptExecutionContext::serviceWorker): Deleted.
* Source/WebCore/inspector/agents/InspectorNetworkAgent.cpp:
(WebCore::InspectorNetworkAgent::enable):
(WebCore::InspectorNetworkAgent::webSocketForRequestId):
(WebCore::webSocketAsScriptValue):
(WebCore::InspectorNetworkAgent::resolveWebSocket):
* Source/WebCore/inspector/agents/InspectorNetworkAgent.h:
* Source/WebCore/inspector/agents/page/PageNetworkAgent.cpp:
(WebCore::PageNetworkAgent::activeWebSockets):
(): Deleted.
* Source/WebCore/inspector/agents/page/PageNetworkAgent.h:
* Source/WebCore/inspector/agents/worker/WorkerNetworkAgent.cpp:
(WebCore::WorkerNetworkAgent::activeWebSockets):
(): Deleted.
* Source/WebCore/inspector/agents/worker/WorkerNetworkAgent.h:
* Source/WebCore/loader/DocumentLoader.cpp:
(WebCore::scriptExecutionContextIdentifierToLoaderMap):
(WebCore::DocumentLoader::fromScriptExecutionContextIdentifier):
(WebCore::DocumentLoader::loadMainResource):
(): Deleted.
* Source/WebCore/loader/archive/Archive.cpp:
(WebCore::Archive::clearAllSubframeArchives):
* Source/WebCore/loader/archive/Archive.h:
* Source/WebCore/page/PageGroup.cpp:
(WebCore::pageGroups):
(WebCore::PageGroup::pageGroup):
* Source/WebCore/workers/service/ServiceWorker.h:
(WebCore::ScriptExecutionContext::serviceWorker):

Canonical link: https://commits.webkit.org/303210@main
